CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.5.1)
+cmake_minimum_required(VERSION 3.16.3)
```

Comment:
Updated CMake minimal version to 3.16.3, as this is the version install on Ubuntu 20.04 from the official package repositories.